### PR TITLE
fix escrow

### PR DESCRIPTION
**Pull request by vicyyn**
init giving error

### CODE REVIEW

File: tests/escrow/programs/escrow/src/lib.rs [hunk 1]
```diff
@@ -102,24 +123,23 @@ pub mod escrow {
...
+    #[account(mut)]
+    pub initializer: Signer<'info>,
     #[account(
         mut,
         constraint = initializer_deposit_token_account.amount >= initializer_amount
     )]
     pub initializer_deposit_token_account: Account<'info, TokenAccount>,
     pub initializer_receive_token_account: Account<'info, TokenAccount>,
-    #[account(init, payer = initializer, space = 8 + EscrowAccount::LEN)]
-    pub escrow_account: Account<'info, EscrowAccount>,
```
Comment (acamill) [inline tests/escrow/programs/escrow/src/lib.rs:114]: I did these change yesterday and launched test in the wrong shell. Reverted locally. Will look into it when I have time. 
And I think it was merged without checking CI.

**Review by acamill — APPROVED** [review]

File: tests/escrow/programs/escrow/src/lib.rs [hunk 2]
```diff
@@ -25,74 +25,95 @@ declare_id!("Fg6PaFpoGXkYsidMpWTK6W2BeZ7FEfcYkg476zPFsLnS");
...
             .to_account_info()
             .key;
-        ctx.accounts.escrow_account.initializer_amount = initializer_amount;
-        ctx.accounts.escrow_account.taker_amount = taker_amount;
+        escrow_account.initializer_amount = initializer_amount;
+        escrow_account.taker_amount = taker_amount;
 
-        let (pda, _bump_seed) = Pubkey::find_program_address(&[ESCROW_PDA_SEED], ctx.program_id);
-        token::set_authority(ctx.accounts.into(), AuthorityType::AccountOwner, Some(pda))?;
+        let (pda, _bump_seed) = Pubkey::find_program_address(&[b"escrow"], ctx.program_id);
```
Comment (acamill) [inline tests/escrow/programs/escrow/src/lib.rs:48]: This should stay out of the init probably, down in the contextes

File: tests/escrow/programs/escrow/src/lib.rs [hunk 3]
```diff
@@ -151,15 +171,16 @@ pub struct CancelEscrow<'info> {
...
         mut,
-        constraint = escrow_account.initializer_key == *initializer.key,
-        constraint = escrow_account.initializer_deposit_token_account == *pda_deposit_token_account.to_account_info().key,
+        constraint = escrow_account.load_mut()?.initializer_key == *initializer.key,
+        constraint = escrow_account.load_mut()?.initializer_deposit_token_account == *pda_deposit_token_account.to_account_info().key,
         close = initializer
     )]
-    pub escrow_account: Account<'info, EscrowAccount>,
-    pub token_program: Program<'info, Token>,
+    pub escrow_account: Loader<'info, EscrowAccount>,
```
Comment (armaniferrante) [inline tests/escrow/programs/escrow/src/lib.rs:178]: Let's not use zero copy for this example as it's not needed.

File: tests/escrow/programs/escrow/src/lib.rs [hunk 4]
```diff
@@ -25,74 +25,95 @@ declare_id!("Fg6PaFpoGXkYsidMpWTK6W2BeZ7FEfcYkg476zPFsLnS");
...
             .key;
-        ctx.accounts.escrow_account.initializer_amount = initializer_amount;
-        ctx.accounts.escrow_account.taker_amount = taker_amount;
+        escrow_account.initializer_amount = initializer_amount;
+        escrow_account.taker_amount = taker_amount;
 
-        let (pda, _bump_seed) = Pubkey::find_program_address(&[ESCROW_PDA_SEED], ctx.program_id);
-        token::set_authority(ctx.accounts.into(), AuthorityType::AccountOwner, Some(pda))?;
+        let (pda, _bump_seed) = Pubkey::find_program_address(&[b"escrow"], ctx.program_id);
+        let cpi_accounts = SetAuthority {
```
Comment (armaniferrante) [inline tests/escrow/programs/escrow/src/lib.rs:49]: It's more idiomatic for this to be in the impl block as it previously was.

**Comment by armaniferrante** [comment]
Escrow changes fixed here https://github.com/project-serum/anchor/pull/767 without reverting the `init` changes.

**Pull request closed without merging**